cost/azure/rightsize_managed_disks/azure_rightsize_managed_disks.pt

```diff
@@ -0,0 +1,1135 @@
+  var ultraSize = 1024
+  while (ultraSize <= 65536) {
+    ultraDiskSizes.push(ultraSize)
+    ultraSize += 1024
+  }
+
+  function findDiskSize(diskTier, diskSizeGB) {
+    var diskSizes = null
+    switch (diskTier) {
+      case "Standard":
+        diskSizes = hddDiskSizes
+        break;
+      case "StandardSSD":
+      case "Premium":
+        diskSizes = ssdDiskSizes
+        break;
+      case "Ultra":
+        diskSizes = ultraDiskSizes
+        break;
+      default:
+        throw "Only Standard HDD, Standard SSD, Premium SSD v1 or Ultra disk tiers are supported."
+    }
+    for (i in diskSizes) {
+      var availableSize = diskSizes[i]
+      if (diskSizeGB <= availableSize) {
+        return availableSize
+      }
+    }
+  }
+
+  function findDiskName(diskTier, diskSizeGB) {
+    diskTier = diskTier.split("_")[0]
+    try {
+      diskSizeGB = findDiskSize(diskTier, diskSizeGB)
+    } catch (err) {
+      return undefined
+    }
+    for (diskName in ds_azure_md_tier_types) {
+      var diskData = ds_azure_md_tier_types[diskName]
+      if (diskData.tier === diskTier && diskData.sizeGiB === diskSizeGB) {
+        return diskName
+      }
+    }
+  }
+
+  function findDiskPrice(diskName, diskRedundancy, diskRegion) {
+    var diskPrice = diskName + "_" + diskRedundancy
+    var pricesByRegion = ds_azure_md_pricing[diskRegion]
+    if (pricesByRegion === undefined) {
+      throw "Region " + diskRegion + " not found at azure_md_pricing.json"
+    }
+    var priceData = pricesByRegion[diskPrice]
+    if (priceData === undefined) {
+      throw "Price for " + diskPrice + " not found at azure_md_pricing.json"
+    }
+    return priceData
+  }
+
+  function findPremSsdV2Price(diskRegion, currentUsedGB, currentUsedIops, currentUsedThroughput) {
+    var premSsdV2Pricing = ds_azure_md_pricing[diskRegion]
+    if (premSsdV2Pricing.PREMIUM_LRS_PROVISIONED_CAPACITY === undefined) { return }
+    if (currentUsedIops <= 3000 && currentUsedThroughput <= 3000) {
+      return premSsdV2Pricing.PREMIUM_LRS_PROVISIONED_CAPACITY.pricePerUnit * 24 * 30 * currentUsedGB
+    }
+    var diskIopsAndThroughputDontFitInDiskSize = true
+    var suggestedDiskSize = currentUsedGB
+    var maxIops = 0
+    var maxThroughput = 0
+    while (diskIopsAndThroughputDontFitInDiskSize) {
+      maxIops = suggestedDiskSize * 500
+      maxThroughput = maxIops * 0.25
+      if (currentUsedIops <= maxIops && currentUsedThroughput <= maxThroughput) {
+        diskIopsAndThroughputDontFitInDiskSize = false
+      }
+      suggestedDiskSize += 1
+    }
+    return {
+      pricePerUnit: Math.round(
+          (premSsdV2Pricing.PREMIUM_LRS_PROVISIONED_CAPACITY.pricePerUnit * 24 * 30 * suggestedDiskSize)
+        + (premSsdV2Pricing.PREMIUM_LRS_PROVISIONED_IOPS.pricePerUnit * 24 * 30 * currentUsedIops)
+        + (premSsdV2Pricing.PREMIUM_LRS_PROVISIONED_THROUGHPUT_MBPS.pricePerUnit * 24 * 30 * currentUsedThroughput)
```

Comment:
If this multiplication is to get the hours of the month, multiply the value by 730 instead of by 24 and then by 30.

Reply:
Thank you @Rangelmv23 ! I addressed this comment :)